src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -27254,14 +27193,33 @@ static class E
     }
 }
 """;
-        var comp = CreateCompilation(src, targetFramework: TargetFramework.Net70);
+        var comp = CreateCompilation(src, targetFramework: TargetFramework.Net70, parseOptions: TestOptions.Regular14);
         comp.VerifyEmitDiagnostics(
             // (2,5): error CS0021: Cannot apply indexing with [] to an expression of type 'C'
             // _ = c[^1];
             Diagnostic(ErrorCode.ERR_BadIndexLHS, "c[^1]").WithArguments("C").WithLocation(2, 5),
-            // (13,20): error CS9282: This member is not allowed in an extension block
+            // (13,20): error CS9500: Extension indexers are not allowed. Please use language version preview or greater to allow.
             //         public int this[int i] => throw null;
-            Diagnostic(ErrorCode.ERR_ExtensionDisallowsMember, "this").WithLocation(13, 20));
+            Diagnostic(ErrorCode.ERR_ExtensionDisallowsIndexerMember, "this").WithArguments("preview").WithLocation(13, 20));
+
+        comp = CreateCompilation(src, targetFramework: TargetFramework.Net70);
+        comp.VerifyEmitDiagnostics(
+            // (2,5): error CS0021: Cannot apply indexing with [] to an expression of type 'C'
```

Comment:
>// (2,5): error CS0021: Cannot apply indexing with [] to an expression of type 'C' [](http://example.com/codeflow?start=12&length=82)

I assume a design decision about this behavior is still pending. Consider adding a PROTOTYPE comment